src/pybind/mgr/dashboard/services/access_control.py

```diff
@@ -515,45 +515,6 @@ def accessdb_config_key(cls, version=None):
             version = cls.VERSION
         return "{}{}".format(cls.ACDB_CONFIG_KEY, version)
 
-    def check_and_update_db(self):
-        logger.debug("Checking for previews DB versions")
-
-        def check_migrate_v0_to_current():
-            # check if there is username/password from previous version
-            username = mgr.get_module_option('username', None)
-            password = mgr.get_module_option('password', None)
-            if username and password:
-                logger.debug("Found single user credentials: user=%s", username)
-                # found user credentials
-                user = self.create_user(username, "", None, None)
-                # password is already hashed, so setting manually
-                user.password = password
-                user.add_roles([ADMIN_ROLE])
-                self.save()
-
-        def check_migrate_v1_to_current():
-            # Check if version 1 exists in the DB and migrate it to current version
-            v1_db = mgr.get_store(self.accessdb_config_key(1))
-            if v1_db:
-                logger.debug("Found database v1 credentials")
-                v1_db = json.loads(v1_db)
-
-                for user, _ in v1_db['users'].items():
-                    v1_db['users'][user]['enabled'] = True
-                    v1_db['users'][user]['pwdExpirationDate'] = None
-                    v1_db['users'][user]['pwdUpdateRequired'] = False
-
-                self.roles = {rn: Role.from_dict(r) for rn, r in v1_db.get('roles', {}).items()}
-                self.users = {un: User.from_dict(u, dict(self.roles, **SYSTEM_ROLES))
-                              for un, u in v1_db.get('users', {}).items()}
-
-                self.save()
-            else:
-                # If version 1 does not exist, check if migration of VERSION "0" needs to be done
-                check_migrate_v0_to_current()
-
-        check_migrate_v1_to_current()
```

Comment:
@votdev @p-se you're right, I think @aaSharma14 shouldn't have deleted the second method of this class...

Please feel free to revert that. We'll merge ASAP.

Reply:
I've created https://tracker.ceph.com/issues/51443 for that and working on it.